MacroChallenge/MacroChallenge/Controller/LevelController.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+import SpriteKit
+
+public final class LevelController {
+    
+    var state: levelState 
+    
+    var controller: LevelProtocol
+    
+    init(state: levelState, controller: LevelProtocol) {
+        self.state = state
+        self.controller = controller
+    }
+    
+    func initState(_ state: levelState) {
+        self.state = state
+    }
+    
+    func update(_ deltaTime: TimeInterval) {
+        switch state {
+            case .cutscene:
+                initCutscene()
+            case .firstWeapon:
+                initFirstWeapon()
+            case .firstMonster:
+                initFirstMonster()
+            case .spaceShipGone:
+                initSpaceShipGone()
+            case .monsterCrew:
+                initMonsterCrew()
+            case .medKit:
+                initMedKit()
+            case .dialog:
+                initDialog()
+            case .findSpaceShip:
+                initFindSpaceShip()
+            case .end:
+                initEnd()
+        }
+    }
+    
+    func initCutscene() {
+        print("")
+    }
+
+    func initFirstWeapon() {
+        print("")
+    }
+
+    func initFirstMonster() {
+        print("")
+    }
+
+    func initSpaceShipGone() {
+        print("")
+    }
+
+    func initMonsterCrew() {
+        print("")
+    }
+
+    func initMedKit() {
+        print("")
+    }
+
+    func initDialog() {
+        print("")
+    }
+
+    func initFindSpaceShip() {
+        print("")
+    }
+
+    func initEnd() {
+        print("")
+    }
+}
+
+enum levelState {
```

Comment:
enum names should be in PascalCase 
```suggestion
enum LevelState {
```

Reply:
PascalCase vai ficar a cabeça da minha rola fdp